File: android-kiosco/app/src/main/java/com/hoteldelvalle/kiosco/Prefs.java
package com.hoteldelvalle.kiosco;

import android.content.Context;
import android.content.SharedPreferences;

public class Prefs {
    private static final String NAME = "kiosko_prefs";
    private static final String KEY_BASE_URL = "base_url";
    private static final String KEY_PIN = "pin";
    private static final String DEFAULT_BASE_URL = "http://68.168.20.219:8000";
    private static final String DEFAULT_PIN = "12345";

    private final SharedPreferences prefs;

    public Prefs(Context context) {
        prefs = context.getSharedPreferences(NAME, Context.MODE_PRIVATE);
    }

    public String getBaseUrl() {
        return prefs.getString(KEY_BASE_URL, DEFAULT_BASE_URL);
    }

    public void setBaseUrl(String url) {
        prefs.edit().putString(KEY_BASE_URL, url).apply();
    }

    public String getPin() {
        return prefs.getString(KEY_PIN, DEFAULT_PIN);
    }

    public void setPin(String pin) {
        prefs.edit().putString(KEY_PIN, pin).apply();
    }

    public boolean isConfigured() {
        return prefs.contains(KEY_BASE_URL);
    }
}

File: android-kiosco/app/src/main/java/com/hoteldelvalle/kiosco/data/Prefs.kt
// Dep: DataStore Preferences 1.1.1
package com.hoteldelvalle.kiosco.data

import android.content.Context
import androidx.datastore.preferences.core.edit
import androidx.datastore.preferences.core.stringPreferencesKey
import androidx.datastore.preferences.preferencesDataStore
import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.map

private val Context.dataStore by preferencesDataStore(name = "kiosko_prefs")

class Prefs(private val context: Context) {

    private val KEY_URL = stringPreferencesKey("server_url")
    private val KEY_PIN = stringPreferencesKey("pin")

    val serverUrl: Flow<String> = context.dataStore.data.map { prefs ->
        prefs[KEY_URL] ?: DEFAULT_URL
    }

    val pin: Flow<String> = context.dataStore.data.map { prefs ->
        prefs[KEY_PIN] ?: DEFAULT_PIN
    }

    suspend fun setUrl(url: String) {
        context.dataStore.edit { it[KEY_URL] = url }
    }

    suspend fun setPin(pin: String) {
        context.dataStore.edit { it[KEY_PIN] = pin }
    }

    companion object {
        const val DEFAULT_URL = "http://68.168.20.219:8000"
        const val DEFAULT_PIN = "12345"
    }
}
